Simpler times - deployment manifests designed for human BOSH operators

drnic opened the pull request:
![simpler-beers](https://user-images.githubusercontent.com/108/33267049-ac18c0fe-d3c3-11e7-9050-1e28a5038e17.jpg)

The goal of this WIP PR is to bring kubo-deployment to the people who already have a BOSH & perhaps CF. Right now it offers two exciting deployment stories:

* "just works" without any variables

    ```
    bosh deploy manifests/cfcr.yml
    ```
* CF routing

    ```
    bosh deploy kubo-deployment/manifests/cfcr.yml \
      -o kubo-deployment/manifests/ops-files/cf-routing.yml \
      -l <(./kubo-deployment/manifests/helper/cf-routing-vars.sh)
    ```

NOTE: due to the continued development of `master` branch and this PR rebasing against it, I've had to fork this PR into two stands.

* To deploy kubo/0.9.0:

    ```
    git clone https://github.com/drnic/kubo-deployment -b stable-0.9.0
    ```

* To deploy kubo/latest (this PR against `master` branch):

    ```
    git clone https://github.com/drnic/kubo-deployment -b cfcr-simpler-times
    ```

The PR also adds sanity tests to operator files against the base manifest using `bin/run_opsfile_test` (thanks to https://github.com/cloudfoundry/cf-deployment !!)

![screen shot 2017-11-29 at 5 49 16 pm](https://user-images.githubusercontent.com/108/33363815-a9737482-d52d-11e7-8049-297a0a4c177e.png)


This PR is not yet feature parity/full documentation/full upgrade of existing wrapper scripts. Since the PR might not be merged without CFCR Core team having written it themselves, I'll leave it for now.

I do love how simple the commands above are now.

iainsproat commented on manifests/cfcr.yml:
```diff
@@ -1,43 +1,59 @@
 - alias: trusty
   os: ubuntu-trusty
-  version: "3468.5"
+  version: 3468.latest
```
The CI pipeline is failing due to the following error:
`Error: No stemcell found at 'https://s3.amazonaws.com/bosh-core-stemcells/google/bosh-stemcell-3468.latest-google-kvm-ubuntu-trusty-go_agent.tgz'.`

drnic commented:
Due to the major refactoring of properties + links in kubo-release master branch, I've decided that this PR will assume kubo-release master branch only. I've got a new branch `stable-0.9.0` for manifests/operators for kub/0.9.0.

I'd like kubo-deployment to have a `develop` -> `release-candidate` -> `master` branch system so that `master` will in future always be manifests for a stable final release

drnic commented:
HURRAY GREEN CI!

I've squashed the 30+ commits into a single commit with a nice commit message.

drnic commented:
I tried to rebase against master but...
```
CONFLICT (content): Merge conflict in src/kubo-deployment-tests/resources/environments/test_gcp_with_service_key/director.yml
Auto-merging src/kubo-deployment-tests/resources/environments/test_gcp_with_cf_and_service_key/director.yml
Auto-merging src/kubo-deployment-tests/resources/environments/test_gcp/director.yml
Auto-merging src/kubo-deployment-tests/resources/environments/test_aws_cf/director.yml
Auto-merging src/kubo-deployment-tests/resources/environments/test_aws/director.yml
Auto-merging src/kubo-deployment-tests/generate_manifest_test.go
CONFLICT (content): Merge conflict in src/kubo-deployment-tests/generate_manifest_test.go
CONFLICT (modify/delete): manifests/ops-files/worker-haproxy.yml deleted in HEAD and modified in simpler times: one lovely base manifest plus PR tests. Version simpler times: one lovely base manifest plus PR tests of manifests/ops-files/worker-haproxy.yml left in tree.
CONFLICT (rename/delete): manifests/ops-files/worker-haproxy-vsphere.yml deleted in HEAD and renamed to manifests/ops-files/iaas/vsphere/worker-haproxy.yml in simpler times: one lovely base manifest plus PR tests. Version simpler times: one lovely base manifest plus PR tests of manifests/ops-files/iaas/vsphere/worker-haproxy.yml left in tree.
Auto-merging manifests/ops-files/iaas/vsphere/master-static-ip.yml
CONFLICT (rename/delete): manifests/ops-files/k8s-haproxy-static-ips-vsphere.yml deleted in HEAD and renamed to manifests/ops-files/iaas/vsphere/cloud-config/k8s-haproxy-static-ips.yml in simpler times: one lovely base manifest plus PR tests. Version simpler times: one lovely base manifest plus PR tests of manifests/ops-files/iaas/vsphere/cloud-config/k8s-haproxy-static-ips.yml left in tree.
CONFLICT (rename/delete): manifests/ops-files/worker-haproxy-openstack.yml deleted in HEAD and renamed to manifests/ops-files/iaas/openstack/worker-haproxy.yml in simpler times: one lovely base manifest plus PR tests. Version simpler times: one lovely base manifest plus PR tests of manifests/ops-files/iaas/openstack/worker-haproxy.yml left in tree.
Auto-merging manifests/ops-files/iaas/aws/cloud-provider.yml
Auto-merging manifests/ops-files/addons-spec.yml
CONFLICT (content): Merge conflict in manifests/ops-files/addons-spec.yml
Auto-merging manifests/cfcr.yml
CONFLICT (content): Merge conflict in manifests/cfcr.yml
Auto-merging configurations/generic/project-config.yml
Auto-merging bin/lib/deploy_utils
CONFLICT (content): Merge conflict in bin/lib/deploy_utils
Auto-merging bin/generate_cloud_config
CONFLICT (content): Merge conflict in bin/generate_cloud_config
```

😢

akshaymankar commented:
Finally, we are done with this!! 🕺 
Only thing that we thought was missing was tests for `cf-routing-vars.sh`. But we left that out as of now. @drnic, if you could write those tests that would be awesome. Thanks again for the PR!

drnic commented:
xoxo

I’ll have a look at the new master branch and see what I can do for cf-routing-vars.sh. I’d probably just rather wait until my bosh links work filters thru routing-release & cf-deployment and then fix the links.yml ops file.

________________________________
From: Akshay Mankar <notifications@github.com>
Sent: Tuesday, December 12, 2017 11:03:17 PM
To: cloudfoundry-incubator/kubo-deployment
Cc: Dr Nic Williams; Mention
Subject: Re: [cloudfoundry-incubator/kubo-deployment] Simpler times - deployment manifests designed for human BOSH operators (#237)


Finally, we are done with this!! 🕺
Only thing that we thought was missing was tests for cf-routing-vars.sh. But we left that out as of now. @drnic<https://github.com/drnic>, if you could write those tests that would be awesome. Thanks again for the PR!

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/cloudfoundry-incubator/kubo-deployment/pull/237#issuecomment-351032064>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AAAAbGr0PFn86cqxI41efygmbRg4uopGks5s_muFgaJpZM4Qrk6W>.
